endpoint.go

```diff
@@ -243,15 +244,15 @@ func (ep *endpoint) KeyPrefix() []string {
 	return []string{datastore.EndpointKeyPrefix, string(n.id)}
 }
 
-func (ep *endpoint) networkIDFromKey(key []string) (types.UUID, error) {
-	// endpoint Key structure : endpoint/network-id/endpoint-id
-	// it's an invalid key if the key doesn't have all the 3 key elements above
-	if key == nil || len(key) < 3 || key[0] != datastore.EndpointKeyPrefix {
-		return types.UUID(""), fmt.Errorf("invalid endpoint key : %v", key)
+func (ep *endpoint) networkIDFromKey(key string) (string, error) {
+	// endpoint Key structure : docker/libnetwork/endpoint/${network-id}/${endpoint-id}
+	// it's an invalid key if the key doesn't have all the 5 key elements above
+	keyElements := strings.Split(key, "/")
+	if !strings.HasPrefix(key, "docker/libnetwork/endpoint/") || len(keyElements) < 5 {
+		return "", fmt.Errorf("invalid endpoint key : %v", key)
 	}
-
 	// network-id is placed at index=1. pls refer to endpoint.Key() method
-	return types.UUID(key[1]), nil
+	return strings.Split(key, "/")[3], nil
```

Comment:
I have plans of changing the key length dependencies (such as 3 or 5) out of objects and make it a datastore specific issue so that we are free to change the key length in datastore without impacting the objects.

We can address this for later.  this is just FYI.